server/api/download.go

```diff
@@ -21,25 +25,45 @@ func (a *API) Download(c *gin.Context) {
 		return
 	}
 	s := payload.ToSnapshot()
-	st, err := a.Metadata.GetSnapshot(ctx, s.Namespace, s.Name, s.Version, s.Latest)
+	data, err := a.download(ctx, s, payload.FullNames)
+	if err != nil {
+		c.Error(err)
+		return
+	}
+	fileName := payload.Version
```

Comment:
Please correct me if I am wrong but I don't see we set `payload.Version` anywhere?

Reply:
Payload.Version come from user payload and it's required field.